hazelcast/src/hazelcast/client/compact.cpp

```diff
@@ -1691,28 +1866,122 @@ schema_writer::build() &&
     return schema{ type_name, std::move(field_definition_map) };
 }
 
+default_schema_service::default_schema_service(spi::ClientContext& context)
+  : retry_pause_millis_{ context.get_client_properties().get_integer(
+      context.get_client_properties().get_invocation_retry_pause_millis()) }
+  , max_put_retry_count_{ context.get_client_properties().get_integer(
+      client_property{ MAX_PUT_RETRY_COUNT, MAX_PUT_RETRY_COUNT_DEFAULT }) }
+  , context_{ context }
+{
+}
+
 schema
 default_schema_service::get(int64_t schemaId)
 {
-    auto ptr = schemas.get(schemaId);
-    if (ptr == nullptr) {
-        // TODO sancar throw schema_does_not_exist;
+    auto ptr = replicateds_.get(schemaId);
+
+    if (!ptr) {
+        throw exception::illegal_state{ "default_schema_service::get",
+                                        "Schema doesn't exist for this type" };
     }
+
     return *ptr;
 }
-void
-default_schema_service::put(const schema& schema_v)
+
+boost::future<void>
+default_schema_service::replicate_schema(schema s)
+{
+    return replicate_schema_attempt(std::move(s));
+}
+
+boost::future<void>
+default_schema_service::replicate_schema_attempt(schema s, int attempts)
+{
+    using hazelcast::client::protocol::ClientMessage;
+    using namespace protocol::codec;
+
+    auto max_retry_count{ max_put_retry_count_ };
+    auto message = send_schema_request_encode(s);
+
+    auto invocation =
+      spi::impl::ClientInvocation::create(context_, message, SERVICE_NAME);
+
+    return invocation->invoke().then(
+      boost::launch::sync,
+      [this, max_retry_count, attempts, s](
```

Comment:
As we are passing this pointer, there is no need for creating and sending max_retry_count to lambda function.